backend/app/routers/authentication.py

```diff
@@ -199,3 +202,39 @@ async def revoke_admin(
             status_code=403,
             detail=f"User {current_username.email} is not an admin. Only admin can revoke admin access.",
         )
+
+
+@router.post("/users/enable_readonly/{useremail}", response_model=UserOut)
+async def enable_readonly_user(
+    useremail: str, current_username=Depends(get_current_user), admin=Depends(get_admin)
+):
+    if admin and current_username.admin:
+        if (user := await UserDB.find_one(UserDB.email == useremail)) is not None:
+            user.read_only_user = True
```

Comment:
I was wondering if we can create a helper function to enable/disable users since the 2 functions is almost exactly same apart from setting
user.read_only_user = True/False
and use the helper function.

Reply:
I tried doing this, but somehow the new function never got called by the front end after I had added it.

This might be related to codegen not working. I had to add it into the services manually, so maybe that is why it does not work.